### PR TITLE
cli: Add alpha support for triggering deploys from cli

### PR DESCRIPTION
**Pull request by ekerfelt**

### CODE REVIEW

File: cli/cmd/encore/deploy.go [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+			if err != nil {
+				cmdutil.Fatalf("no app found. Run deploy inside an encore app directory or specify the app with --app")
+			}
+			appSlug, err = appfile.Slug(appRoot)
+			if err != nil {
+				cmdutil.Fatalf("no app found. Run deploy inside an encore app directory or specify the app with --app")
+			}
+		}
+		if envName == "" {
+			envName = "@primary"
```
Comment (eandre) [inline cli/cmd/encore/deploy.go:49]: Is this a good idea? Feels a bit risky to not be explicit about which environment to deploy to.

File: cli/cmd/encore/deploy.go [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+)
+
+var deployAppCmd = &cobra.Command{
+	Use:                   "deploy --commit COMMIT_SHA | --branch BRANCH_NAME",
+	Short:                 "Deploy an Encore app to a cloud environment",
+	DisableFlagsInUseLine: true,
+	Hidden:                true,
+	Run: func(c *cobra.Command, args []string) {
+		if commit != "" {
+			hb, err := hex.DecodeString(commit)
```
Comment (eandre) [inline cli/cmd/encore/deploy.go:33]: Thoughts on using `git rev-parse` here? Even if we require the commit to be pushed it still seems like a nice convenience. Maybe not worth it?

File: cli/cmd/encore/deploy.go [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+	},
+}
+
+func init() {
+	alphaCmd.AddCommand(deployAppCmd)
+	deployAppCmd.Flags().StringVar(&appSlug, "app", "", "app slug to deploy to (default current app)")
+	deployAppCmd.Flags().StringVarP(&envName, "env", "e", "", "environment to deploy to (default primary env)")
+	deployAppCmd.Flags().StringVarP(&commit, "commit", "c", "", "commit to deploy")
+	deployAppCmd.Flags().StringVarP(&branch, "branch", "b", "", "branch to deploy")
+	deployAppCmd.Flags().StringVarP(&format, "format", "f", "text", "format of output. One of (text, json)")
```
Comment (eandre) [inline cli/cmd/encore/deploy.go:98]: We have `cmdutil.Oneof` for this use case which does some niceties like autocomplete and validation

File: cli/cmd/encore/deploy.go [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+		}
+	},
+}
+
+func init() {
+	alphaCmd.AddCommand(deployAppCmd)
+	deployAppCmd.Flags().StringVar(&appSlug, "app", "", "app slug to deploy to (default current app)")
+	deployAppCmd.Flags().StringVarP(&envName, "env", "e", "", "environment to deploy to (default primary env)")
+	deployAppCmd.Flags().StringVarP(&commit, "commit", "c", "", "commit to deploy")
+	deployAppCmd.Flags().StringVarP(&branch, "branch", "b", "", "branch to deploy")
```
Comment (eandre) [inline cli/cmd/encore/deploy.go:97]: The shorthands seem a bit esoteric; consider removing?

File: cli/cmd/encore/deploy.go (hunk 1 above)
Comment (ekerfelt) [reply cli/cmd/encore/deploy.go:49]: Yeah, you're probably right. Was striving to be consistent with other commands.

File: cli/cmd/encore/deploy.go (hunk 2 above)
Comment (ekerfelt) [reply cli/cmd/encore/deploy.go:33]: Yeah, maybe we should. Can't assume that git is installed or the command is run from within a local repository. But I suppose I could run it if --app is not set and git is available. On fail, I could parse the the revision specifier manually and submit it as a commit sha if it's a 20 byte hex, or a branch if it's not.

Comment (ekerfelt) [reply cli/cmd/encore/deploy.go:33]: Probably rename the flag to revision then?

Comment (eandre) [reply cli/cmd/encore/deploy.go:33]: I guess that's fair, let's just keep it as-is and not spend more time on it

**Review by eandre — APPROVED** [review]

**Pull request merged**